src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -231,16 +361,27 @@ bool AutomatedRssDownloader::isRssDownloaderEnabled() const
     return ui->checkEnableDownloader->isChecked();
 }
 
-void AutomatedRssDownloader::updateRuleDefinitionBox()
+void AutomatedRssDownloader::updateRuleDefinitionBox(QListWidgetItem *selected)
 {
+    disconnectRuleFeedSlots();
+
     qDebug() << Q_FUNC_INFO;
     // Save previous rule first
     saveEditedRule();
     // Update rule definition box
     const QList<QListWidgetItem *> selection = ui->listRules->selectedItems();
-    if (selection.count() == 1) {
-        m_editedRule = selection.first();
-        Rss::DownloadRulePtr rule = getCurrentRule();
+
+    if (!selected && (selection.count() == 1))
```

Comment:
No - no `selected` parameter was passed into the function, and there is exactly one object selected in the UI, we set that as the selection.

The next branch `if (selected)` is only followed if there is exactly one selected item.

Note that this code has already been merged as part of PR #6326.